src/server/datachan.rs

```diff
@@ -267,6 +267,29 @@ where
 {
     // We introduce a block scope here to keep the lock on the session minimal. We basically copy the needed info
     // out and then unlock.
+
+    match socket.peer_addr() {
+        Ok(datachan_addr) => {
+            let controlcahn_ip = session_arc.lock().await.source.ip();
+            if controlcahn_ip != datachan_addr.ip() {
+                if let Err(err) = socket.shutdown(std::net::Shutdown::Both) {
```

Comment:
Also log when there is no shutdown error.

Reply:
added